### PR TITLE
MULE-14529: XMLSecureFactories is setting unsupported properties at f…

### PR DESCRIPTION
**Pull request by facundovs**
…actories creation.

### CODE REVIEW

File: core/src/main/java/org/mule/util/xmlsecurity/DefaultXMLSecureFactories.java [hunk 1]
```diff
@@ -19,6 +20,7 @@
 import javax.xml.transform.TransformerFactory;
 import javax.xml.validation.SchemaFactory;
 import javax.xml.validation.Validator;
+import javax.xml.XMLConstants;
```
Comment (aiannucci) [inline core/src/main/java/org/mule/util/xmlsecurity/DefaultXMLSecureFactories.java]: Unused import

File: core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    private final TransformerFactory transformerFactoryWrapper = mock(TransformerFactory.class);
+
+    @Test
+    public void validatorProperties() throws Exception
+    {
+        SetPropertyAnswer setPropertyAnswer = new SetPropertyAnswer(schemaFactory.newSchema().newValidator());
+        doAnswer(setPropertyAnswer).when(validatorWrapper).setProperty(anyString(), anyObject());
+        defaultXMLSecureFactories.configureValidator(validatorWrapper);
+        assertThat(setPropertyAnswer.exception, is(nullValue()));
+        verify(validatorWrapper, times(2)).setProperty(anyString(), anyObject());
```
Comment (aiannucci) [inline core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java]: How does this test validate that the correct properties are set?

File: core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @Test
+    public void schemaFactoryProperties() throws Exception
+    {
+        SetPropertyAnswer setPropertyAnswer = new SetPropertyAnswer(schemaFactory);
+        doAnswer(setPropertyAnswer).when(schemaFactoryWrapper).setProperty(anyString(), anyObject());
+        defaultXMLSecureFactories.configureSchemaFactory(schemaFactoryWrapper);
+        assertThat(setPropertyAnswer.exception, is(nullValue()));
+        verify(schemaFactoryWrapper, times(2)).setProperty(anyString(), anyObject());
```
Comment (aiannucci) [inline core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java]: Same as before

File: core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @Test
+    public void transformerFactoryProperties()
+    {
+        SetPropertyAnswer setPropertyAnswer = new SetPropertyAnswer(transformerFactory);
+        doAnswer(setPropertyAnswer).when(transformerFactoryWrapper).setAttribute(anyString(), anyObject());
+        defaultXMLSecureFactories.configureTransformerFactory(transformerFactoryWrapper);
+        assertThat(setPropertyAnswer.exception, is(nullValue()));
+        verify(transformerFactoryWrapper, times(2)).setAttribute(anyString(), anyObject());
```
Comment (aiannucci) [inline core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java]: Same as before

File: core/src/main/java/org/mule/util/xmlsecurity/DefaultXMLSecureFactories.java (hunk 1 above)
Comment (facundovs) [reply core/src/main/java/org/mule/util/xmlsecurity/DefaultXMLSecureFactories.java]: Not really. This class is being used in javadocs.

File: core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java (hunk 2 above)
Comment (facundovs) [reply core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java]: This mock object has an answer for the setProperty() method that invokes the real method and verifies any exception was thrown.

**Pull request merged**